File: fibonacci/fibonacci.py
import datetime, sys

n = int(sys.argv[1])

def fib_pd(pos):
  mem = [0]*(pos+1)
  for i in range(pos+1):
    f=0
    if i<=2:
      f=1
    else:
      f=mem[i-1]+mem[i-2]
    mem[i]=f
  return mem[pos]

beginTime = datetime.datetime.now()

# print(fib_pd(n))
fib_pd(n)

totalTime = datetime.datetime.now() - beginTime
print("\nIn Python - Fibonnaci series in position %d took: %sms\n" % (n, str(totalTime)))

File: fibonacci/fibonacci.go
package main

import (
	"fmt"
	"math/big"
	"os"
	"strconv"
	"time"
)

func main() {
	arg := os.Args[1]
	n, _ := strconv.Atoi(arg)

	start := time.Now()

	// fmt.Println(fibPd(n))
	fibPd(n)

	end := time.Now()
	totalTime := end.Sub(start)

	fmt.Printf("\nIn GO - Fibonnaci series in position %d took: %s\n", n, totalTime)
}

func fibPd(pos int) *big.Int {
	mem := make([]*big.Int, pos+1)
	for i := 0; i <= pos; i++ {
		var f = big.NewInt(0)
		if i <= 2 {
			f.SetUint64(1)
		} else {
			f = f.Add(mem[i-1], mem[i-2])
		}
		mem[i] = f
	}
	return mem[pos]
}
